include/dlaf/common/single_threaded_omp.h

```diff
@@ -0,0 +1,32 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2022, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+
+// TODO: internal?
+namespace dlaf::common::internal {
+class [[nodiscard]] SingleThreadedOmpScope {
```

Comment:
Not fully satisfied with the name as the main goal is to reduce blas threads.
What about `SingleThreadBlasScope`? Or `SingleThreadScope`?

Reply:
Indeed, I'm also not entirely happy with having `Omp` in the name, especially since it also sets MKL threads (which is not necessarily using OpenMP).

I like `SingleThread(ed)Blas(Scope)` as well. `SingleThreadScope` sounds slightly too generic for my taste, but it's also ok if you prefer that.

Reply:
It's now called `SingleThreadedBlasScope`.